elide-core/src/main/java/com/yahoo/elide/core/EntityDictionary.java

```diff
@@ -76,8 +108,30 @@ protected EntityBinding entityBinding(Class<?> entityClass) {
      * @param entityClass the entity class
      * @return binding class
      */
-    public String getBinding(Class<?> entityClass) {
-        return entityBinding(entityClass).jsonApi;
+    public String getJsonAliasFor(Class<?> entityClass) {
+        return getEntityBinding(entityClass).jsonApiType;
+    }
+
+    public boolean entityHasChecksForPermission(Class<?> resourceClass, Class<? extends Annotation> annotationClass) {
+        EntityBinding binding = getEntityBinding(resourceClass);
+        return binding.entityPermissions.hasChecksForPermission(annotationClass);
+    }
+
+    public PermissionDefinition getPermissionsForClass(Class<?> resourceClass,
+                                                       Class<? extends Annotation> annotationClass) {
+        EntityBinding binding = getEntityBinding(resourceClass);
+        return binding.entityPermissions.getClassChecksForPermission(annotationClass);
+    }
+
+    public PermissionDefinition getPermissionsForField(Class<?> resourceClass,
+                                                       String field,
+                                                       Class<? extends Annotation> annotationClass) {
+        EntityBinding binding = getEntityBinding(resourceClass);
+        return binding.entityPermissions.getFieldChecksForPermission(field, annotationClass);
+    }
+
+    public Class<? extends Check> getCheck(String checkIdentifier) {
+        return checkNames.get(checkIdentifier);
```

Comment:
If checkNames don't contain the identifier, we should throw a runtime exception here with meaningful description.